src/server/svr_comm.c

```diff
@@ -310,6 +310,112 @@ send_nodestat_req(enum msvr_stats_type type)
 	mtfd = -1;
 }
 
+/**
+ * @brief broadcast discard job request to peer servers
+ * 
+ * @param[in] jobid - job id
+ * @param execvnode - job's exec vnode
+ * @param exechost - job's exec host
+ * @param rver - job run version
+ * 
+ * @return int 
+ * @retval 0: success
+ * @retval !0: failure
+ */
+int
+ps_send_discard(char *jobid, char *execvnode, char *exechost, int rver)
+{
+	int mtfd = -1;
+	int rc;
+
+	if (!exechost)
+		return -1;
+
+	mtfd = open_ps_mtfd();
+	if (mtfd == -1)
+		return -1;
+
+	rc = ps_compose(mtfd, PS_DISCARD_JOB);
+	if (rc != DIS_SUCCESS)
+		goto err;
+
+	if ((rc = diswcs(mtfd, jobid, strlen(jobid))) != 0)
+		goto err;
+
+	if ((rc = diswcs(mtfd, execvnode, strlen(execvnode))) != 0)
+		goto err;
+
+	if ((rc = diswcs(mtfd, exechost, strlen(exechost))) != 0)
+		goto err;
```

Comment:
We already sent exec_vnode to the peer server with PS_RSC_UPDATE, do we really need to send it again?

Reply:
PS_RSC_UPDATE and PS_DISCARD_JOB are independent messages. It is not necessary that a resource update will be sent when a discard job is sent and vice versa.